### PR TITLE
Add note about automatic window title to FileDialog documentation.

### PR DESCRIPTION
**Pull request by dominiks**
For #38896

### CODE REVIEW

File: doc/classes/FileDialog.xml [hunk 1]
```diff
@@ -4,7 +4,7 @@
 		Dialog for selecting files or directories in the filesystem.
 	</brief_description>
 	<description>
-		FileDialog is a preset dialog used to choose files and directories in the filesystem. It supports filter masks.
+		FileDialog is a preset dialog used to choose files and directories in the filesystem. It supports filter masks. The FileDialog automatically sets its window title according to the [code]mode[/code]. Disable this with [code]mode_overrides_title[/code] if you want to use a custom window title.
```
Comment (Calinou) [inline doc/classes/FileDialog.xml:7]: Use property references so they can be clicked:

```suggestion
		FileDialog is a preset dialog used to choose files and directories in the filesystem. It supports filter masks. The FileDialog automatically sets its window title according to the [member file_mode. If you want to use a custom title, disable this by setting [member mode_overrides_title] to [code]false[/code].
```

**Review by Calinou — CHANGES_REQUESTED** [review]

File: doc/classes/FileDialog.xml [hunk 2]
```diff
@@ -4,7 +4,7 @@
 		Dialog for selecting files or directories in the filesystem.
 	</brief_description>
 	<description>
-		FileDialog is a preset dialog used to choose files and directories in the filesystem. It supports filter masks.
+		FileDialog is a preset dialog used to choose files and directories in the filesystem. It supports filter masks. The FileDialog automatically sets its window title according to the [member file_mode. If you want to use a custom title, disable this by setting [member mode_overrides_title] to [code]false[/code].
```
Comment (Calinou) [inline doc/classes/FileDialog.xml:7]: Sorry, I made a typo in the previous suggestion:

```suggestion
		FileDialog is a preset dialog used to choose files and directories in the filesystem. It supports filter masks. The FileDialog automatically sets its window title according to the [member file_mode]. If you want to use a custom title, disable this by setting [member mode_overrides_title] to [code]false[/code].
```

**Review by Calinou — CHANGES_REQUESTED** [review]

**Comment by YeldhamDev** [comment]
Please squash your commits: https://docs.godotengine.org/en/stable/community/contributing/pr_workflow.html#the-interactive-rebase

**Review by YeldhamDev — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.2.2.

**Pull request merged**